operators/src/source/gdal_source/loading_info.rs

```diff
@@ -54,6 +53,129 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+        time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    ) -> Self {
+        let mut sorted_time_steps = time_steps;
+        sorted_time_steps.sort_by_key(TimeInterval::start);
+
+        GdalMetadataFixedTimes {
+            time_steps: sorted_time_steps,
+            params,
+            result_descriptor,
+            time_placeholders,
+        }
+    }
+
+    fn create_temporal_slice(
+        &self,
+        step_interval: TimeInterval,
+        empty: bool,
+    ) -> Result<GdalLoadingInfoTemporalSlice> {
+        if empty {
+            Ok(GdalLoadingInfoTemporalSlice {
+                time: step_interval,
+                params: None,
+            })
+        } else {
+            self.params
+                .replace_time_placeholders(&self.time_placeholders, step_interval)
+                .map(|loading_info_part_params| GdalLoadingInfoTemporalSlice {
+                    time: step_interval,
+                    params: Some(loading_info_part_params),
+                })
+        }
+    }
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetadataFixedTimes
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        if self.time_steps.is_empty() {
+            return Err(Error::EmptyTimeSteps);
+        }
+
+        let valid_duration = TimeInterval::new(
+            self.time_steps[0].start(),
+            self.time_steps[self.time_steps.len() - 1].start(),
+        )?;
+
+        let parts = match valid_duration.intersect(&query.time_interval) {
+            Some(time_interval) => {
+                let mut loading_info_parts: Vec<GdalLoadingInfoTemporalSlice> = Vec::new();
+
+                if query.time_interval.start() < time_interval.start() {
+                    let gap =
+                        TimeInterval::new(query.time_interval.start(), time_interval.start())?;
+
+                    loading_info_parts.push(self.create_temporal_slice(gap, true)?);
+                }
+
+                for (index, time_position) in self.time_steps.iter().enumerate() {
+                    if index > 0 && self.time_steps[index - 1].end() < time_position.start() {
+                        let time_step_gap = TimeInterval::new(
+                            self.time_steps[index - 1].end(),
+                            time_position.start(),
+                        )?;
+
+                        if let Some(gap_step_interval) = time_interval.intersect(&time_step_gap) {
+                            loading_info_parts
+                                .push(self.create_temporal_slice(gap_step_interval, true)?);
+                        }
+                    };
+
+                    if time_position.start() > time_interval.end() {
+                        break;
+                    }
+
+                    if let Some(step_interval) = time_interval.intersect_inclusive(time_position) {
```

Comment:
why does it have to be `inclusive` here? If it is really necessary, then please move this method into `impl GdalMetadataFixedTimes` we don't want this used anywhere else in the code because it violates our time semantics

Reply:
It is necessary to have `inclusive` here to include the last `time_step` when it is a `TimeInstance` and at the end of the `time_interval`. For example if the `time_interval` is [10, 20) and the last `time_step` within `time_interval` is [20, 20), then it is necessary to  have `inclusive` to include the last `time_step`.

Reply:
the `end` of a time interval doesn't belong to it. So the instance [20, 20) in your example shouldn't be included right?. It's like if I am querying January as [1.1.2000, 1.2.2000) I wouldn't want an Instant on the first of February returned.

Reply:
Okay, I will change it back to `intersect` without `inclusive`. I was under the assumption that all TimeInstances in the query interval have to be returned.